metaflow/flowspec.py

```diff
@@ -100,6 +100,13 @@ def __init__(self, use_cli=True):
         self._graph = FlowGraph(self.__class__)
         self._steps = [getattr(self, node.name) for node in self._graph]
 
+        # This must be set before calling cli.main() below (or specifically, add_custom_parameters)
+        parameters.parameters = [
+            getattr(self, a)
+            for a in dir(self)
```

Comment:
There's already a function known as `_get_parameters`, we can probably re-use that instead?

Reply:
something like: `parameters.parameters = [p for _, p in self._get_parameters()]`